app/jobs/upload_users_job.rb

```diff
@@ -0,0 +1,24 @@
+class UploadUsersJob < ApplicationJob
+  USER_FIELDS = [:user_name, :last_name, :first_name, :id_number, :email].freeze
+
+  def self.on_complete_js(_status)
+    'window.location.reload.bind(window.location)'
+  end
+
+  def perform(user_class, data, encoding)
+    progress.total = data.lines.count
+    user_class.transaction do
+      MarkusCsv.parse(data, encoding: encoding, skip_blanks: true, row_sep: :auto) do |row|
+        user_data = {}
+        USER_FIELDS.each_with_index do |field, index|
+          row_index = EndUser::CSV_ORDER.index(field) || index
+          user_data[field] = row[row_index]&.strip
+        end
+        next if user_data[:user_name].blank? || user_data[:last_name].blank? || user_data[:first_name].blank?
```

Comment:
This check isn't necessary. If these fields are blank there will be a validation error below, which will be reported as an error by the job---and displayed as a flash message in the UI, which you should double-check to see.